app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comment
+  has_many :likes
+
+  def last_five_comments
+    Comment.all.where("post_id = #{id}").limit(5).order(created_at: :desc)
+  end
+
+  def update_posts
+    author.increment!(:posts_counter)
+  end
```

Comment:
[Optional] I suggest you use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `update_posts` method every time a new post has been saved to database, then make the `update_posts` method private

For example
```
after_save :update_posts

  private

  def update_posts
    author.increment!(:posts_counter)
  end
```

Note: This can be also be done for updating methods like `update_post_comments` and `update_post_likes`

Reply:
Thanks. I'm on it.